feat: badge component

harshitadaswani opened the pull request:
added profile badges and number badges in the badge component

harshitadaswani committed:
feat: badge component

added profile badges and number badges

rkapoor10 commented on components/badge.css:
```diff
@@ -0,0 +1,46 @@
+}
+
+.offline {
+    background-color: grey;
```
use css variables for colors

rkapoor10 commented on components/badge.css:
```diff
@@ -0,0 +1,46 @@
+}
+
+.dnd {
+    background-color: red;
```
same as above

rkapoor10 commented on components/badge.html:
```diff
@@ -0,0 +1,98 @@
+            </aside>
+
+        <main class="main-container">
+            <h1 class="h1-title">Badge</h1>
```
if h1-title gives colors name it accordingly

harshitadaswani replied:
h1-title is a utility CSS class having styling related to the title elements

rkapoor10 commented on components/badge.html:
```diff
@@ -0,0 +1,98 @@
+        <main class="main-container">
+            <h1 class="h1-title">Badge</h1>
+            <p class="border-bottom">Badges are small circles, positioned either at top-right or bottom-right of the parent component. Badge can be used to display numbers, online / offline status, depending on where they are used.</p>
+            <h2 class="h2-title">Profile Badge</h2>
```
same as above

harshitadaswani replied:
h1-title is a utility CSS class having styling related to the title elements

rkapoor10 commented on components/badge.html:
```diff
@@ -0,0 +1,98 @@
+                </div>            
+            </div>
+            <iframe src="https://www.thiscodeworks.com/embed/61faef71b783be0015bbaf6c" style="width: 95%; height: 20rem;" class="code-snippet" frameborder="0"></iframe>
+            <div class="border-bottom"></div>
```
what is the use of this empty div?

rkapoor10 commented on components/badge.html:
```diff
@@ -0,0 +1,98 @@
+            <div class="example-div br-l">
+                <div class="badge">
+                    <i class="fas fa-envelope icon"></i>
+                    <div class="number-badge">10</div>
```
separate out classes for different properties...if possible.

rkapoor10 commented on stylesheets/base.css:
```diff
@@ -69,7 +69,7 @@ img,iframe {
 
 [class^="f"] {
-    font-size: inherit;
+    font-size: 2rem;
```
find a way around...don't use fixed font size.

rkapoor10 reviewed (changes requested):
Refactor the code according to the given comments. Rest all is good.

harshitadaswani committed:
fix: reviews suggested by the reviewer

- added space to example div
- clubbed all the classes together having common properties
- fixed font-size of font-awesome icons
- defined variable names to colors used in badge component and used them appropriately

mariyasada commented on components/avatar.css:
```diff
@@ -26,7 +26,6 @@
     padding: var(--l-size);
     display: flex;
     flex-flow: row wrap;
```
delete that unnecessary comment

harshitadaswani replied:
already deleted! thanks

sks-21 reviewed (approved):
Everything looks nice. Good to merge !

rkapoor10 reviewed (approved):
Great! good to merge.
